### PR TITLE
Document instrumentation

### PR DESCRIPTION
**Pull request by gustawlippa**
This PR adds documentation for instrumentation. There are also a number of small changes related with it that update the documentation and fix errors.
Instrumentation changes:
- Added `Instrumentation` page under `Configuration`, which documents the configuration options.
- Prometheus endpoint is documented in HTTP listeners.
- In `Logging and Monitoring`, Prometheus is described as an Exometer alternative, as well as GraphQL endpoint is mentioned. This file could be maybe revisited and updated later, maybe changed to a tutorial on setting up metrics/instrumentation.
- Metrics in `Operation and maintenance/Metrics` have been updated, mainly with the new Prometheus naming scheme.
- Update metrics descriptions in modules documentation.

Other changes:
- When clicking on a header, a link to this header is copied,
- Automatic links to next/previous pages is added at the bottom of the page. I think it makes sense, for example at the intrudactory sections. Sometimes it leads to a bit surprising links (for example from the `Configuration` page back to `MAM` page), but I think it's not a big issue, and the addition is worth it.
- Port syntax was incorrectly described in a few places.
- `c2s_message_processing_time` had an incorrect measurement declared - `byte_size`. It was supposed to be `time`, but then, the name would be `c2s_message_processing_time_time` for Exometer. So, the metric was renamed to `c2s_message_processed`, and the measurement fixed to `time`.
- Links to tabbed content now have more readable headers - the name of the tab, instead of numbers.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/234591) / elasticsearch_and_cassandra_mnesia / 53b50fd8ceba90f776687348678efd6dd75efe27
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4355/234591/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4355/234591/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4096824fc0ecee.2024-08-06_07.39.21/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/234892) / elasticsearch_and_cassandra_mnesia / 3d67e13950a576d2a80bc139d989b39d3fd39ba8
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4355/234892/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4355/234892/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%40f44fe51055ef.2024-08-09_14.55.37/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/234979) / elasticsearch_and_cassandra_mnesia / 5e71f487d9a77dba5b50824da9f138fd9e3d9706
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4355/234979/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4355/234979/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%403cbd250702bd.2024-08-12_09.46.30/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235012) / elasticsearch_and_cassandra_mnesia / 252a0dab2c669f0374fbdb274933e73d452ca902
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4355/235012/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4355/235012/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%408ab88b002256.2024-08-12_10.12.16/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235171) / elasticsearch_and_cassandra_mnesia / 30190ee6825a69966932c042bcaff7b1f4a37195
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4355/235171/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4355/235171/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%406d228220df74.2024-08-12_14.54.04/index.html)
**OK: 461** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: doc/operation-and-maintenance/MongooseIM-metrics.md [hunk 1]
```diff
@@ -170,111 +258,231 @@ Metrics specific to an extension, e.g. Message Archive Management, are described
...
-* `[HostType, backends, auth, does_user_exist]`
+Prometheus metrics have a `host_type` label associated with these metrics, as well as a `function` label describing the backend action.
+Since Exometer doesn't support labels, the host types and backend actions are part of the metric names.
+
+=== "Prometheus"
+
+    | Metric name | Type | Description |
+    | ----------- | ---- | ----------- |
+    | `BackendModule_count` | counter | Number of calls (spiral metric), incremented for *every* call (even a failed one). |
+    | `BackendModule_time` | histogram | Successful operation times. |
```
Comment (jacekwegr) [inline doc/operation-and-maintenance/MongooseIM-metrics.md:442]: This is just my opinion, but I think the description would be a bit clearer if phrased this way
```suggestion
    | `BackendModule_time` | histogram | Times of successful operations. |
```

File: doc/modules/mod_event_pusher_http.md [hunk 2]
```diff
@@ -110,10 +110,23 @@ Below is an example of what the body of an HTTP POST request can look like:
...
+Prometheus metrics have a `host_type` label associated with these metrics.
+Since Exometer doesn't support labels, the host types, or word `global`, are part of the metric names, depending on the [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global) option.
+
+=== "Prometheus"
+
+    | Name | Type | Description (when it gets incremented) |
+    | ---- | ---- | -------------------------------------- |
+    | `mod_event_pusher_http_sent_count` | counter | An HTTP notification is sent successfully. |
+    | `mod_event_pusher_http_sent_failure_count` | counter | An HTTP notification failed. |
+    | `mod_event_pusher_http_sent_response_time` | histogram | Does not include timings of failed requests. |
```
Comment (jacekwegr) [inline doc/modules/mod_event_pusher_http.md:122]: I think it would be a good idea to first describe what the metric does and then add the comment about failed requests
```suggestion
    | `mod_event_pusher_http_sent_response_time` | histogram | Time taken to send HTTP notification. Does not include timings of failed requests. |
```

**Review by jacekwegr — COMMENTED** [review]
Looks great! I can see a lot of effort was put into updating and improving the documentation. I just have a few small comments.

File: doc/modules/mod_event_pusher_http.md (hunk 2 above)
Comment (gustawlippa) [reply doc/modules/mod_event_pusher_http.md:122]: I agree. I must admit, with module metrics, I mostly copied the old description.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235237) / elasticsearch_and_cassandra_mnesia / 9636520c96bd03c345e39355b7be5ba8bad0c403
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4355/235237/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4355/235237/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%40dd5f15fe1da2.2024-08-13_11.49.05/index.html)
**OK: 461** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235279) / elasticsearch_and_cassandra_mnesia / e0e819c54925da8ebe226f7ef086fd25961556a5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4355/235279/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4355/235279/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%40bb76ca7588e6.2024-08-13_11.53.37/index.html)
**OK: 461** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Review by jacekwegr — APPROVED** [review]
Looks good. Thank you for the changes!

File: doc/operation-and-maintenance/Logging-&-monitoring.md [hunk 3]
```diff
@@ -59,45 +59,36 @@ Additionally, WombatOAM provides interfaces to feed the data to other OAM tools
...
+
+#### Prometheus
+
+MongooseIM exposes metrics in the Prometheus format by default.
+More information about configuration can be found in [the HTTP listeners section](../listeners/listen-http.md#handler-types-prometheus-mongoose_prometheus_handler).
+
+#### Exometer
 
 MongooseIM uses [a fork of Exometer library](https://github.com/esl/exometer_core) for collecting metrics.
 Exometer has many plug-in reporters that can send metrics to external services. We maintain [exometer_report_graphite](https://github.com/esl/exometer_report_graphite) and [exometer_report_statsd](https://github.com/esl/exometer_report_statsd) for Graphite and StatsD respectively.
```
Comment (chrzaszcz) [inline doc/operation-and-maintenance/Logging-&-monitoring.md:83]: Ok, so the problem is that we don't support StatsD anymore in instrumentation. We would need to add support. Maybe we need to add this functionality? But we would need to add some tests, as there are none. I think I could add a story for this.

**Review by chrzaszcz — COMMENTED** [review]
I found some places where the docs can be improved. I think we could address them in a follow-up PR.

File: doc/operation-and-maintenance/Logging-&-monitoring.md (hunk 3 above)
Comment (gustawlippa) [reply doc/operation-and-maintenance/Logging-&-monitoring.md:83]: Yes, I've noticed this too as well, and wasn't sure if I should remove this line. I will try to reword the next sentence maybe to be less misleading.
As a sidenote, I think we should in general revisit this doc page in general and make sure it is still correct. I'd say maybe even change it to a tutorial along the lines of "How to set up metrics for MongooseIM?", with differences between Exometer/Prometheus explained and up to a Grafana dashboard. We have a task to create a Grafana dashboard for MIM, which could be completed beforehand.

File: doc/configuration/instrumentation.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
+This section is used to configure MongooseIM instrumentation.
+It is a system of executing events when something of interest happens in the server.
+They are mainly used for the purpose of metrics.
+
+Instrumentation events are acted upon by handlers. Available instrumentation handlers are:
+
+* `prometheus` - exposes a metrics endpoint for [Prometheus](https://prometheus.io/).
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/configuration/instrumentation.md:7]: Fixed in https://github.com/esl/MongooseIM/commit/6b436b42a1e53592482491a4ad3fd0c6d28ca320

File: doc/modules/mod_vcard.md [hunk 5]
```diff
@@ -127,10 +127,32 @@ An array of search fields, which values should be Base64-encoded by MongooseIM b
...
+Prometheus metrics have a `host_type` and `function` labels associated with these metrics.
+Since Exometer doesn't support labels, the function as well as the host types, or word `global`, are part of the metric names, depending on the [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global) option.
+
+Backend in the action name can be either `rdbms`, `ldap` or `mnesia`.
+
+=== "Prometheus"
+
+    | Backend action | Type | Function | Description (when it gets incremented) |
+    | -------------- | ---- | -------- | -------------------------------------- |
+    | `mod_private_Backend_count` | counter | `set_vcard` | A vCard is set in a database. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_vcard.md:142]: Fixed in https://github.com/esl/MongooseIM/commit/0c74dd031c02cae294b6d07dde4082ff1bc37f0a.

File: doc/operation-and-maintenance/Logging-&-monitoring.md (hunk 3 above)
Comment (gustawlippa) [reply doc/operation-and-maintenance/Logging-&-monitoring.md:83]: I reworded this paragraph here: https://github.com/esl/MongooseIM/commit/4447d7ef46e06c5a8100543942521a57f69a34af

File: doc/configuration/instrumentation.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+## Exometer options
+
+General options for the Exometer reporter:
+
+### `instrumentation.exometer.all_metrics_are_global`
+* **Syntax:** boolean
+* **Default:** `false`
+* **Example:** `all_metrics_are_global = true`
+
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/configuration/instrumentation.md:31]: Fixed in https://github.com/esl/MongooseIM/commit/a4f6dc377cf2890288ee2e9b25c2af6a5683c63a.

File: doc/configuration/instrumentation.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+A prefix to prepend all metric names with before they are sent to the graphite server.
+
+### `instrumentation.exometer.report.graphite.env_prefix`
+* **Syntax:** string
+* **Default:** no default
+* **Example:** `env_prefix = "GRAPHITE_METRICS_PREFIX"`
+
+Specifies an environmental variable name from which an additional prefix will be taken.
+In case both `prefix` and `env_prefix` are defined, it will be placed before the `prefix` and separated with a dot.
+
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/configuration/instrumentation.md:87]: Fixed in https://github.com/esl/MongooseIM/commit/b8969b570648e049b301ceac6f49eb49709ed51c.

File: doc/modules/mod_muc_light.md [hunk 8]
```diff
@@ -176,18 +176,72 @@ Each `config_schema` item is a TOML table with the following keys:
...
+Prometheus metrics have a `host_type` and `function` labels associated with these metrics.
+Since Exometer doesn't support labels, the function as well as the host types, or word `global`, are part of the metric names, depending on the [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global) option.
+
+Backend in the action name can be either `rdbms` or `mnesia`.
+
+=== "Prometheus"
+
+    | Backend action | Type | Function | Description (when it gets incremented) |
+    | -------------- | ---- | -------- | -------------------------------------- |
+    | `mod_muc_light_Backend_count` | counter | `create_room` | A new room is stored in a DB. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_muc_light.md:191]: Fixed in https://github.com/esl/MongooseIM/commit/4aaa2271ada74017cd8677c4d29de67667cbb84b.

File: doc/modules/mod_last.md [hunk 9]
```diff
@@ -29,9 +29,32 @@ Storage backend.
...
+=== "Prometheus"
+
+    | Backend action | Type | Function | Description (when it gets incremented) |
+    | -------------- | ---- | -------- | -------------------------------------- |
+    | `mod_last_Backend_count` | counter | `get_last` | A timestamp is fetched from the database. |
+    | `mod_last_Backend_time` | histogram | `get_last` | Time spent fetching a timestamp from the database. |
+    | `mod_last_Backend_count` | counter | `set_last_info` |  A timestamp is stored in the database. |
+    | `mod_last_Backend_time` | histogram | `set_last_info` | Time spent storing a timestamp in the database. |
+    | `mod_last_Backend_count` | counter | `session_cleanup` | A session is cleaned up from the database. |
+    | `mod_last_Backend_time` | histogram | `session_cleanup` | Time spent cleaning up a from the database. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_last.md:49]: Fixed in https://github.com/esl/MongooseIM/commit/7575e72132411b7bb06837c1c2fb1c64819513ad.

File: doc/modules/mod_offline.md [hunk 10]
```diff
@@ -46,9 +46,28 @@ to disable this error message.
...
+Prometheus metrics have a `host_type` and `function` labels associated with these metrics.
+Since Exometer doesn't support labels, the function as well as the host types, or word `global`, are part of the metric names, depending on the [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global) option.
+
+Backend in the action name can be either `rdbms` or `mnesia`.
+
+=== "Prometheus"
+
+    | Backend action | Type | Function | Description (when it gets incremented) |
+    | -------------- | ---- | -------- | -------------------------------------- |
+    | `mod_last_Backend_count` | counter | `pop_messages` | Offline messages for a user are retrieved and deleted from a DB. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_offline.md:61]: Fixed in https://github.com/esl/MongooseIM/commit/aabaeebee821244c90034cbaf6474fc3f63f031d.

File: doc/modules/mod_ping.md [hunk 11]
```diff
@@ -53,8 +53,21 @@ Strategy to handle incoming stanzas. For details, please refer to
...
+Prometheus metrics have a `host_type` label associated with these metrics.
+Since Exometer doesn't support labels, the host types, or word `global`, are part of the metric names, depending on the [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global) option.
+
+=== "Prometheus"
+
+    | Name | Type | Description (when it gets incremented) |
+    |------|------|----------------------------------------|
+    | `mod_ping_response_count` | counter | Client responds to a ping. |
+    | `mod_ping_response_time` | histogram | Response times (doesn't include timeouts). |
+    | `mod_ping_response_count` | counter | Ping request timeouts without a response from client. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_ping.md:65]: FIxed in https://github.com/esl/MongooseIM/commit/07a87d256e8fbbde52964b92941b0dcf83020ab0.

File: doc/modules/mod_pubsub.md [hunk 12]
```diff
@@ -234,66 +234,198 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
...
+    | `[HostType, mod_pubsub_cache_Backend, get_last_item, count]` | counter | Last item from a node is fetched. |
+    | `[HostType, mod_pubsub_cache_Backend, get_last_item, time]` | histogram | Time to fetch the last item from a node. |
+
+#### Database backend
+
+=== "Prometheus"
+
+    | Backend action | Type | Function | Description (when it gets incremented) |
+    | -------------- | ---- | -------- | -------------------------------------- |
+    | `mod_pubsub_cache_Backend_count` | counter | `get_state` | User's state for a specific node is fetched. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_pubsub.md:347]: Fixed in https://github.com/esl/MongooseIM/commit/30e16c69bec205238ca10df3b5b727953c2e50da.

File: doc/modules/mod_roster.md [hunk 13]
```diff
@@ -43,17 +43,52 @@ Improves performance but should be disabled, when shared rosters are used.
...
+Prometheus metrics have a `host_type` and `function` labels associated with these metrics.
+Since Exometer doesn't support labels, the function as well as the host types, or word `global`, are part of the metric names, depending on the [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global) option.
+
+Backend in the action name can be either `rdbms` or `mnesia`.
+
+=== "Prometheus"
+
+    | Backend action | Type | Function | Description (when it gets incremented) |
+    | -------------- | ---- | -------- | -------------------------------------- |
+    | `mod_private_Backend_count` | counter | `read_roster_version` | Version of a user's roster is retrieved. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_roster.md:58]: Fixed in https://github.com/esl/MongooseIM/commit/03ecfe69881ab8faa544c8525bf5f62cfb17a681.

File: doc/modules/mod_last.md [hunk 14]
```diff
@@ -29,9 +29,32 @@ Storage backend.
...
+
+=== "Exometer"
+
+    | Backend action | Type | Description (when it gets incremented) |
+    | -------------- | ---- | -------------------------------------- |
+    | `[HostType, mod_last_Backend, get_last, count]` | counter | A timestamp is fetched from the database. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_last.md:55]: Fixed in https://github.com/esl/MongooseIM/commit/df0da1e00563cbd8bf34226c419285b0c8275d52.

File: doc/modules/mod_pubsub.md [hunk 15]
```diff
@@ -234,66 +234,198 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
...
+    | `mod_pubsub_cache_Backend_count` | counter | `get_subnodes_tree` | Full tree of subnodes of a node is fetched. |
+    | `mod_pubsub_cache_Backend_time` | histogram | `get_subnodes_tree` | Time to fetch the full tree of subnodes of a node. |
+    | `mod_pubsub_cache_Backend_count` | counter | `get_parentnodes_tree` | All parents of a node are fetched. |
+    | `mod_pubsub_cache_Backend_time` | histogram | `get_parentnodes_tree` | Time to fetch all parents of a node. |
+
+=== "Exometer"
+
+    | Backend action | Type | Description (when it gets incremented) |
+    | -------------- | ---- | -------------------------------------- |
+    | `[HostType, mod_pubsub_cache_Backend, get_state, count]` | counter | User's state for a specific node is fetched. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_pubsub.md:392]: Fixed in https://github.com/esl/MongooseIM/commit/df0da1e00563cbd8bf34226c419285b0c8275d52.

File: doc/modules/mod_pubsub.md [hunk 16]
```diff
@@ -234,66 +234,198 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
...
+    | `mod_pubsub_cache_Backend_count` | counter | `delete_last_item` | Last item is deleted from a node. |
+    | `mod_pubsub_cache_Backend_time` | histogram | `delete_last_item` | Time to delete last item from a node. |
+    | `mod_pubsub_cache_Backend_count` | counter | `get_last_item` | Last item from a node is fetched. |
+    | `mod_pubsub_cache_Backend_time` | histogram | `get_last_item` | Time to fetch the last item from a node. |
+
+=== "Exometer"
+
+    | Backend action | Type | Description (when it gets incremented) |
+    | -------------- | ---- | -------------------------------------- |
+    | `[HostType, mod_pubsub_cache_Backend, upsert_last_item, count]` | counter | Last item is upserted into a node. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_pubsub.md:334]: Fixed in https://github.com/esl/MongooseIM/commit/df0da1e00563cbd8bf34226c419285b0c8275d52.

File: doc/modules/mod_offline.md [hunk 17]
```diff
@@ -46,9 +46,28 @@ to disable this error message.
...
+    | `mod_last_Backend_time` | histogram | `pop_messages` | Time spent retrieving and deleting offline messages for a user from a DB. |
+    | `mod_last_Backend_count` | counter | `write_messages` |  New offline messages to a user are written in a DB. |
+    | `mod_last_Backend_time` | histogram | `write_messages` | Time spent writing a new offline messages to a user in a DB. |
+
+=== "Exometer"
+
+    | Backend action | Type | Description (when it gets incremented) |
+    | -------------- | ---- | -------------------------------------- |
+    | `[HostType, mod_last_Backend, pop_messages, count]` | counter | Offline messages for a user are retrieved and deleted from a DB. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_offline.md:70]: Fixed in https://github.com/esl/MongooseIM/commit/df0da1e00563cbd8bf34226c419285b0c8275d52.

File: doc/operation-and-maintenance/MongooseIM-metrics.md [hunk 18]
```diff
@@ -170,111 +258,231 @@ Metrics specific to an extension, e.g. Message Archive Management, are described
...
-| `[global, system_up_time, seconds]` | value | Node uptime. |
-| `[global, mnesia_info, running_db_nodes]` | value | A number of nodes in a MongooseIM cluster seen by a given MongooseIM node (based on Mnesia). For CETS, use `[global, cets_info, joined_nodes]` instead. |
-| `[global, system_tcp_ports, count]` | value | A number of open tcp connections. This should relate to the number of connected sessions and databases, as well as federations and http requests. A constantly growing value might indicate a connection leak. |
-| `[global, system_process_queue_lengths, total]` | probe | The total number of incoming messages queued in the Erlang processes. It is a good indicator of an overloaded system: if too many messages are queued at the same time, the system is most likely overloaded with incoming data. |
-| `[global, system_dist_data, Metric]` | gauge | Network stats for Erlang distributed communication. `Metric` can be `recv_oct`, `recv_cnt`, `recv_max`, `send_oct`, `send_max`, `send_cnt`, `send_pend` or `connections`. |
+=== "Prometheus"
+
+    | Name | Type | Description (when it gets incremented) |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/operation-and-maintenance/MongooseIM-metrics.md:263]: Removed in https://github.com/esl/MongooseIM/commit/5774c7f04457ffc168784afeb36c0ec970bef5be.

File: doc/operation-and-maintenance/MongooseIM-metrics.md [hunk 19]
```diff
@@ -170,111 +258,231 @@ Metrics specific to an extension, e.g. Message Archive Management, are described
...
 
 ### Pool metrics
 
 For RDBMS global pool defined, an instance of these metrics are available.
+Prometheus metrics have a `pool_tag` label associated with these metrics.
+Since Exometer doesn't support labels, the tags are part of the metric names.
+
+=== "Prometheus"
+
+    | Name | Type | Description (when it gets incremented) |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/operation-and-maintenance/MongooseIM-metrics.md:389]: Removed in https://github.com/esl/MongooseIM/commit/5774c7f04457ffc168784afeb36c0ec970bef5be.

File: doc/modules/mod_muc_light.md [hunk 20]
```diff
@@ -176,18 +176,72 @@ Each `config_schema` item is a TOML table with the following keys:
...
+    | `mod_muc_light_Backend_count` | counter | `get_aff_users` | Affiliated users list is fetched from a DB. |
+    | `mod_muc_light_Backend_time`  | histogram | `get_aff_users` | Time to fetch affiliated users list from a DB. |
+    | `mod_muc_light_Backend_count` | counter | `modify_aff_users` | Affiliations in a room are updated in a DB. |
+    | `mod_muc_light_Backend_time`  | histogram | `modify_aff_users` | Time to update affiliations in a room in a DB. |
+
+=== "Exometer"
+
+    | Backend action | Type | Description (when it gets incremented) |
+    | -------------- | ---- | -------------------------------------- |
+    | `[HostType, mod_muc_light_Backend, create_room,  count]` | counter   | A new room is stored in a DB. |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_muc_light.md:222]: FIxed in https://github.com/esl/MongooseIM/commit/df0da1e00563cbd8bf34226c419285b0c8275d52.

File: doc/modules/mod_muc.md [hunk 21]
```diff
@@ -488,10 +488,25 @@ If the server returns something else, an error presence will be sent back to the
...
+Since Exometer doesn't support labels, the host types, or word `global`, are part of the metric names, depending on the [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global) option.
+
+=== "Prometheus"
+
+  | Name | Type | Description (when it gets incremented) |
+  | ---- | ---- | -------------------------------------- |
+  | `mod_muc_deep_hibernations_count` | counter | A room process is stopped (applies only to persistent rooms). |
+  | `mod_muc_process_recreations_count` | counter | A room process is recreated from a persisted state. |
+  | `mod_muc_hibernations_count` | counter | A room process becomes hibernated (garbage collected and put in wait state). |
+  | `mod_muc_rooms_hibernated` | gauge | How many rooms are in hibernated state. Does not include rooms in "deep hibernation". |
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply doc/modules/mod_muc.md:501]: I've added a small note to the probes that I found here: https://github.com/esl/MongooseIM/commit/5774c7f04457ffc168784afeb36c0ec970bef5be

**Comment by gustawlippa** [comment]
I tried to address the issues found after the review in https://github.com/esl/MongooseIM/pull/4360
